R/get.samples.R

```diff
@@ -46,19 +57,27 @@ get.samples <- function(catch, lf, al, tresh.al = 2, tresh.lf = 2, period.unit =
     
     period.unit <- match.arg(period.unit)
     
-    # 2) Minor calculations and variable creation
-    options(dplyr.summarise.inform = FALSE) # https://www.tidyverse.org/blog/2020/05/dplyr-1-0-0-last-minute-additions/
-    lf <- lf %>% 
-          group_by_at(lfcol[1:5]) %>% 
-          mutate(prop = n/sum(n)) %>% # proportion of each length in each sample
-          as.data.frame()
+    if(prob.al<0 | prob.al>1) warning("prob.al should be restricted to 0-1")
     
+    # 2) Minor calculations and variable creation
+    id <- as.factor(apply(lf[,lfcol[1:5]],1,paste,collapse=""))
+    lf$prop <- ave(lf$n, id, FUN = function(z) z/sum(z))
     lf$date <- with(lf, ymd(paste(year, period, '01')))
     al$date <- with(al, ymd(paste(year, period, '01')))
     
     my.levels <- sapply(c('period', 'region', 'gear'), function(x) sort(unique(c(lf[, x], al[, x])))) # unique levels of each group
     
-    # 3) For each line of catch, i.e. for each combination of year-period-region-gear, get lf and al data to use in caa calculations
+    # 3) Global alk (alg): from all years and all ages available.
+    alg <- table(al$length,al$age)
+    alg <- prop.table(alg,1)
+    alg <- cbind(length=as.numeric(rownames(alg)),data.frame(rbind(alg)))
+    
+    le <- expand.grid(length=seq(min(alg$length),max(alg$length),min(diff(alg$length))))
```

Comment:
not sure about the 3rd argument. Why not just 1?

Reply:
min(diff)) just calculated bin width, which might be 5cm for one species and 1 for another.